vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -737,10 +715,28 @@ std::vector<ChakraObjectRef> ChakraRuntime::ToChakraObjectRefs(const facebook::j
   return result;
 }
 
-ChakraObjectRef ChakraRuntime::GetProperty(const ChakraObjectRef &obj, const ChakraObjectRef &id) {
-  JsValueRef result = JS_INVALID_REFERENCE;
-  VerifyJsErrorElseThrow(JsGetProperty(obj, id, &result));
-  return ChakraObjectRef(result);
+void ChakraRuntime::SetUpMemoryTracker() {
+  // Note: We must not use VerifyJsErrorElseThrow here because this function is
+  // called in ChakraRuntime's constructor.
+
+  if (m_args->m_memoryLimit) {
+    size_t memoryLimit = *(m_args->m_memoryLimit);
+    if (memoryLimit == 0) {
+      throw facebook::jsi::JSINativeException("ChakraRuntime cannot have a memory limit of zero bytes.");
+    }
+    VerifyChakraErrorElseThrow(JsSetRuntimeMemoryLimit(m_runtime, memoryLimit));
+  }
+
+  if (m_args->m_memoryTracker) {
+    std::shared_ptr<facebook::react::MemoryTracker> &memoryTracker = m_args->m_memoryTracker;
```

Comment:
>std::shared_ptr< [](start = 4, length = 16)

This is super super nitty, but this might be clearer as a ref to the underlying MemoryTracker instead of a ref to the shared_ptr.